### PR TITLE
Refactor merged line creation in UcteImporter to solve inconsistent data

### PR DESCRIPTION
**Pull request by murgeyseb**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
Currently, when importing merged line from UCTE files, the ID of the line is generated based on the IDs of both dangling lines that compose it.
In order to always have the same ID, independently from the order of the dangling lines, they are sorted before creating the merged line ID.

However, there is no consistency between the order of the dangling lines in the generated ID and side one and side two terminals of the line.

**What is the new behavior (if this is a feature change)?**
The reordering is still done, but now the created line as consistent connection sides.

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java [hunk 1]
```diff
@@ -867,76 +864,84 @@ private void mergeXnodeDanglingLines(UcteNetwork ucteNetwork, Network network) {
 
         Set<DanglingLine> danglingLinesToProcess = Sets.newHashSet(network.getDanglingLines());
         while (!danglingLinesToProcess.isEmpty()) {
-            DanglingLine dl1 = danglingLinesToProcess.iterator().next();
-            DanglingLine dl2 = getMatchingDanglingLine(dl1, danglingLinesByXnodeCode);
+            DanglingLine dlToProcess = danglingLinesToProcess.iterator().next();
+            DanglingLine dlMatchingDlToProcess = getMatchingDanglingLine(dlToProcess, danglingLinesByXnodeCode);
```
Comment (mathbagu) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: It's a detail, but the name of this variable should be `matchingDlToProcess`

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java [hunk 2]
```diff
@@ -867,76 +864,84 @@ private void mergeXnodeDanglingLines(UcteNetwork ucteNetwork, Network network) {
...
+        // lexical sort to always end up with same merge line id
+        String mergeLineId = dlAtSideOne.getId() + " + " + dlAtSideTwo.getId();
+
+        // create XNODE merge extension
+        float rdp = (float) (dlAtSideOne.getR() / (dlAtSideOne.getR() + dlAtSideTwo.getR()));
+        float xdp = (float) (dlAtSideOne.getX() / (dlAtSideOne.getX() + dlAtSideTwo.getX()));
+        double xnodeP1 = dlAtSideOne.getP0();
+        double xnodeQ1 = dlAtSideOne.getQ0();
+        double xnodeP2 = dlAtSideTwo.getP0();
+        double xnodeQ2 = dlAtSideTwo.getQ0();
```
Comment (mathbagu) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: Maybe you should respect your own style by removing this variable used once and call `dlAtSideOne.getP0()` (and the others) in the adder.

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java (hunk 1 above)
Comment (murgeyseb) [reply ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: Well, you can also consider that it is the dangling line that matches the dangling line to process.
Which one do you prefer ?

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java (hunk 2 above)
Comment (murgeyseb) [reply ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: They are used twice (also in the merged Xnode extension). Do I really change it ?

Comment (mathbagu) [reply ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java]: I didn't notice that these variables were used twice. Keep it like this

**Pull request merged**